### ISSUE TITLE
Cannot build conda packages for B-Tax release 0.2.3

### ISSUE THREAD

**Issue by martinholmer** [open-source-economics/B-Tax#221]
I've been using B-Tax release 0.2.1 to test the Package-Builder (P-B) tool's [new version](https://github.com/open-source-economics/Package-Builder/pull/112#issue-229976249).  I've not been able to test using B-Tax release 0.2.2 because it set an **exact pin** of 1.15.1 for `taxcalc` in the `conda.recipe/meta.yaml` file.  There are no `taxcalc` packages less than 0.18.0 available on the Anaconda channel `ospc`, so it was not surprising that I couldn't make btax 0.2.2 packages.  
My first question: how were the conda packages for B-Tax release 0.2.2 made?

Now for the main topic of this issue.  The new P-B tool still makes packages for B-Tax release 0.2.1, but cannot make packages for the new B-Tax release 0.2.3.  I'm not sure why this is the case, but I'm curious about the following change that was made to the `conda.recipe/meta.yaml` file on Feb 23, 2018, in PR #203:
<img width="1174" alt="screen shot 2018-11-11 at 5 59 01 pm" src="https://user-images.githubusercontent.com/12170745/48319405-b1e56700-e5db-11e8-8d24-ae67b7a59396.png">

I don't understand why this change was made.  Tax-Calculator and other models that use Tax-Calculator (including OG-USA and Behavioral-Responses) follow the convention that was eliminated in B-Tax PR #203.  The conversation in PR #203 is not very helpful in explaining the reasoning behind the change:

<img width="1357" alt="screen shot 2018-11-11 at 5 59 41 pm" src="https://user-images.githubusercontent.com/12170745/48319447-0ee11d00-e5dc-11e8-948a-a21738a36ce6.png">

Not only is there no rationale for the change, but the PR was merged even though the tests failed.

@jdebacker,  Have you been able to make conda packages for B-Tax release 0.2.3?  

Have you read the recommendations for specifying the `environment.yml` and `conda.recipe/meta.yaml` files when a model depends on `taxcalc` that were contained in [this comment](https://github.com/open-source-economics/Behavioral-Responses/pull/18#issue-228848719)?

Can your follow the dependency-specification recommendations in that comment and change back to `version: 0.0.0` in the `conda.recipe/meta.yaml` file and create a new release 0.2.4?   I'm hoping those changes will make it possible to build conda packages for B-Tax.  Otherwise, B-Tax will be stuck at 0.2.2 in PolicyBrain and for other users of the B-Tax packages.

@MattHJensen @hdoupe

**Comment by martinholmer** [comment]
As an addendum to the first comment in B-Tax issue #221, here are the results (tail end of a lot of screen output) of making conda packages for the latest (0.5.12) release of OG-USA using the new Package-Builder tool:
```
. . .
Using Anaconda API: https://api.anaconda.org
Using "pslmodels" as upload username
Processing 'pkgbld_output/win-64/ogusa-0.5.12-py36_0.tar.bz2'
Detecting file type...
File type is "conda"
Extracting conda package attributes for upload
Creating package "ogusa"
Creating release "0.5.12"
Uploading file "pslmodels/ogusa/0.5.12/win-64/ogusa-0.5.12-py36_0.tar.bz2"
 uploaded 696 of 696Kb: 100.00% ETA: 0.0 minutes
Upload complete

conda package located at:
https://anaconda.org/pslmodels/ogusa


real	3m5.138s
user	1m20.972s
sys	0m30.544s

iMac mrh$ conda search -c PSLmodels ogusa
Loading channels: done
# Name                  Version           Build  Channel             
ogusa                    0.5.12          py36_0  PSLmodels           
```

@jdebacker @rickecon @MattHJensen @hdoupe 

P.S.  You can try this by installing the new Package-Builder tool and using it as follows:
```
iMac mrh$ conda search -c PSLmodels pkgbld
Loading channels: done
# Name                  Version           Build  Channel             
pkgbld                    0.9.0          py36_0  PSLmodels           

iMac mrh$ conda install -c PSLmodels pkgbld --yes

iMac mrh$ cat release-ogusa.py
import pkgbld as pb
pb.release('OG-USA', 'ogusa', '0.5.12')

iMac mrh$ time python release-ogusa.py
```

**Comment by jdebacker** [comment]
> I'm not sure why this is the case, but I'm curious about the following change that was made to the conda.recipe/meta.yaml file on Feb 23, 2018, in PR #203

> I don't understand why this change was made. 

@hdoupe might better recall, but I thought we were trying to automatically pull the version number from the git tag.  There was a change to `meta.yml` on that same day that had set the version to 0.4 (not sure why).  I recall lots of issues trying to build the package around that time, so were were probably changing things and merging quickly, tagging, and testing another build.  At some point we must have given up since you note that a 0.2.2 package can't be built.

> @jdebacker, Have you been able to make conda packages for B-Tax release 0.2.3?

No.  I've never successfully built a conda package for B-Tax or anything else.

> Have you read the recommendations for specifying the environment.yml and conda.recipe/meta.yaml files when a model depends on taxcalc that were contained in this comment?

I hadn't.

> Can your follow the dependency-specification recommendations in that comment and change back to version: 0.0.0 in the conda.recipe/meta.yaml file and create a new release 0.2.4?

Yes.  I just opened PR #222.  I'm happy to make any changes that will help you toward successful build.

**Comment by jdebacker** [comment]
@martinholmer in example above, do I need to create the `cat release-ogusa.py` script?  Then I just put those two lines of code in it?

**Comment by jdebacker** [comment]
> Not really sure what your question is. I've already tested making packages for OG-USA release 0.5.12.

Right, but I was trying to create a B-Tax package and replacing 'ogusa' with 'btax' in your example didn't work.  So I tried to do exactly what you showed to see if I could build an OG-USA package and that didn't work either.

> Have you followed the suggestion I made here in B-Tax issue #223? Once you commit those changes and create a new release 0.2.5

Yes, those changes have been committed, merged, and 0.2.5 released.

@martinholmer

**Comment by martinholmer** [comment]
Issue #221 has been resolved in B-Tax release 0.2.5
